src/modules/BuilderPlace/context/BuilderPlaceContext.tsx

```diff
@@ -1,35 +1,137 @@
+import { createContext, ReactNode, useEffect, useMemo, useState } from 'react';
 import { IBuilderPlace } from '../types';
-import TalentLayerContext from '../../../context/talentLayer';
+import { iBuilderPlaceContext, IUser } from '../../../types';
+import { useChainId } from '../../../hooks/useChainId';
+import { useAccount } from 'wagmi';
+import { getCompletionScores, ICompletionScores } from '../../../utils/profile';
+import { TalentLayerClient } from '@talentlayer/client';
+import { getUserById } from '../../../queries/users';
+import { toast } from 'react-toastify';
 
-const BuilderPlaceContext = createContext<{
-  builderPlace?: IBuilderPlace;
-  isBuilderPlaceOwner: boolean;
-}>({
+const BuilderPlaceContext = createContext<iBuilderPlaceContext>({
+  loading: true,
+  isActiveDelegate: false,
+  refreshData: async () => {
+    return false;
+  },
   builderPlace: undefined,
   isBuilderPlaceOwner: false,
 });
 
 const BuilderPlaceProvider = ({ data, children }: { data: IBuilderPlace; children: ReactNode }) => {
-  const { user } = useContext(TalentLayerContext);
+  const chainId = useChainId();
+  const [user, setUser] = useState<IUser | undefined>();
+  const account = useAccount();
+  const [isActiveDelegate, setIsActiveDelegate] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [completionScores, setCompletionScores] = useState<ICompletionScores | undefined>();
+  const [talentLayerClient, setTalentLayerClient] = useState<TalentLayerClient>();
   const [builderPlace, setBuilderPlace] = useState<IBuilderPlace | undefined>();
   const [isBuilderPlaceOwner, setIsBuilderPlaceOwner] = useState<boolean>(false);
 
+  // automatically switch to the default chain is the current one is not part of the config
   useEffect(() => {
-    if (!data) return;
+    const talentLayerClient = new TalentLayerClient({
+      chainId: process.env.NEXT_PUBLIC_DEFAULT_CHAIN_ID as unknown as number,
+      ipfsConfig: {
+        clientId: process.env.NEXT_PUBLIC_INFURA_ID as string,
+        clientSecret: process.env.NEXT_PUBLIC_INFURA_SECRET as string,
+        baseUrl: process.env.NEXT_PUBLIC_IPFS_WRITE_URL as string,
+      },
+      platformId: parseInt(process.env.NEXT_PUBLIC_PLATFORM_ID as string),
+      signatureApiUrl: process.env.NEXT_PUBLIC_SIGNATURE_API_URL as string,
+    });
+    setTalentLayerClient(talentLayerClient);
+  }, [account.address]);
 
-    const isBuilderPlaceOwner = data?.owners?.some(
-      owner => owner.toLocaleLowerCase() === user?.address.toLocaleLowerCase(),
-    );
+  const fetchData = async () => {
+    if (!data || !data.ownerTalentLayerId || !account.isConnected || !talentLayerClient) {
+      setLoading(false);
+      return false;
+    }
 
-    setIsBuilderPlaceOwner(isBuilderPlaceOwner || false);
-    setBuilderPlace(data);
-  }, [data, user]);
+    try {
+      const userResponse = await getUserById(chainId, data.ownerTalentLayerId);
```

Comment:
it's not efficient to do that on frontend side and it adds a new "loading" step for the frontend. As we got everything we need, we can just move the graph query on server side after getting the BP from db